src/uwtools/rocoto.py

```diff
@@ -0,0 +1,44 @@
+"""
+Support for writing a Rocoto XML using J2Templating
+"""
+
+from uwtools.j2template import J2Template
+from uwtools.config import YAMLConfig
+
+# private functions
+def _add_jobname(tasks: dict) -> None:
+    """
+    Add the jobname entry for all the tasks in the workflow.
+
+    :param tasks: Dict of tasks in workflow.
+    """
+    if not isinstance(tasks, dict):
+        return
+    for task, task_settings in tasks.items():
+        task_type = task.split("_", maxsplit=1)[0]
+        if task_type == "task":
+            # Use the provided attribute if it is present, otherwise use
+            # the name in the key
+            tasks[task]["jobname"] = \
+                task_settings.get("attrs", {}).get("name") or \
+                task.split("_", maxsplit=1)[1]
+        elif task_type == "metatask":
+            _add_jobname(task_settings)
+
+# Main function
+def write_rocoto_xml(input_yaml: str, input_template: str, rendered_output: str) -> None:
+    """
+    Main entry point.
+
+    :param input_yaml: Yaml file provided by the user.
```

Comment:
I'd recommend
```
:param input_yaml: Path to YAML input file.
```
because 1. The other two path argument descriptions all start (helpfully) with "Path to", 2. `YAML` is an acronym and so should be in all-caps, and 3. "the user" is not a useful concept here -- there may be no user, for example via a call from a unit test or some other programmatic use case.

Reply:
I agree with this. It makes it simpler when take a quick glance at the code.